File: PCube-API/domain/expense_account.py
class ExpenseAccount:

    def __init__(self):
        self.id = -1
        self.name = ""
        self.parent_id = -1
        self.parent_name = ""
        self.child = {}
        

    def asDictionary(self):
        return {
            "id": self.id,
            "name": self.name,            
            "parent_id": self.parent_id,
            "parent_name": self.parent_name,
            "childs": self.child       
        }

File: client/src/app/models/expense-account.ts
import {DataTreeFetcher}  from './utils/DataTreeFetcher';

export class ExpenseAccountItem {
    id : number;
    name : string;
    parent_id: number;
    child: ExpenseAccountItem[];
    nomAffichage: string;
    nbLignesDeTemps: number;

    constructor(expenseAccountResponse?: any) {
        this.id = expenseAccountResponse && expenseAccountResponse.id || "";
        this.name = expenseAccountResponse && expenseAccountResponse.name || "";
        this.parent_id = expenseAccountResponse && expenseAccountResponse.parent_id || "";
        this.nomAffichage = expenseAccountResponse && expenseAccountResponse.nomAffichage || "";
        this.child = expenseAccountResponse && expenseAccountResponse.child || [];
        this.nbLignesDeTemps = expenseAccountResponse && expenseAccountResponse.nbLignesDeTemps || "";
    }

    static fetchItemFromResponse(response): ExpenseAccountItem {
        let expenseAccount = new ExpenseAccountItem(response);
        expenseAccount.parent_id = response.ExpenseAccountId;
        return expenseAccount;
    }
}